examples/react/w3console/src/share.tsx

```diff
@@ -0,0 +1,89 @@
+import { ChangeEvent, useState } from 'react'
+import { useKeyring } from '@w3ui/react-keyring'
+import * as DID from '@ipld/dag-ucan/did'
+import { CarWriter } from '@ipld/car/writer'
+import type { PropsWithChildren } from 'react'
+import type { Delegation } from '@ucanto/interface'
+
+function Header (props: PropsWithChildren): JSX.Element {
+  return <h3 className='font-semibold text-xs font-mono uppercase tracking-wider mb-4 text-gray-400'>{props.children}</h3>
+}
+
+export async function toCarBlob (delegation: Delegation): Promise<Blob> {
+  const { writer, out } = CarWriter.create()
+  for (const block of delegation.export()) {
+    // @ts-expect-error
+    void writer.put(block)
+  }
+  void writer.close()
+
+  const carParts = []
+  for await (const chunk of out) {
+    carParts.push(chunk)
+  }
+  const car = new Blob(carParts, {
+    type: 'vnd.ipld.car'
+  })
+  return car
+}
+
+export function SpaceShare (): JSX.Element {
+  const [, { createDelegation }] = useKeyring()
+  const [value, setValue] = useState('')
+  const [downloadUrl, setDownloadUrl] = useState('')
+
+  async function makeDownloadLink (input: string): Promise<void> {
+    let audience
+    try {
+      audience = DID.parse(input.trim())
+    } catch (err) {
+      setDownloadUrl('')
```

Comment:
do we want to stash this error somewhere so we can tell the user why the DID they pasted is wrong or is there some other mechanism for that? either way not urgent, but probably useful for quality of life at some point

Reply:
Yes, I'll circle back in another PR and improve this!